ext/date/php_date.c

```diff
@@ -307,9 +309,25 @@ static zend_object_handlers date_object_handlers_timezone;
 static zend_object_handlers date_object_handlers_interval;
 static zend_object_handlers date_object_handlers_period;
 
-#define DATE_CHECK_INITIALIZED(member, class_name) \
+static void date_throw_error_with_parent_name(zend_class_entry *ce)
+{
+	if (ce->type == ZEND_INTERNAL_CLASS) {
+		zend_throw_error(date_ce_date_object_error, "The %s object has not been correctly initialized by its constructor by calling parent::__construct()", ZSTR_VAL(ce->name));
+	} else {
+		zend_class_entry *ce_ptr = ce;
+		while (ce_ptr && ce_ptr->parent && ce_ptr->type == ZEND_USER_CLASS) {
+			ce_ptr = ce_ptr->parent;
+		}
+		if (ce_ptr->type != ZEND_INTERNAL_CLASS) {
+			zend_throw_error(date_ce_date_object_error, "The %s object has not been correctly initialized by its constructor by calling parent::__construct()", ZSTR_VAL(ce->name));
+		}
+		zend_throw_error(date_ce_date_object_error, "The %s object (inheriting %s) has not been correctly initialized by its constructor by calling parent::__construct()", ZSTR_VAL(ce->name), ZSTR_VAL(ce_ptr->name));
```

Comment:
nit: fwiw I'm not sure I'm curious about the parent class name, it's usually easy to find out from the name or at worst case from the signature of the child.